### PR TITLE
Changes to deploy staging

### PR DESCRIPTION
**Pull request by omar-selo**
Jira card: [RTW-95](https://warthogs.atlassian.net/browse/RTW-95)

This PR includes all changes I had to make to be able to deploy staging successfully.

These changes include:
- Replacing traefik with nginx-ingress-integrator
- Using api url passed from the charm integration on the frontend
- Creating GH actions to release api and frontend charms
- Switching to Postgres version 14 as 15 isn't available as a charm. This also needed a db migration as we were using a feature from Postgres 15 before
- Fixing a few bugs on charms
- Making appropriate changes on Terraform
- Updating readme

I apologize for the big PR but didn't want to open it until I was sure all problems have been dealt with.

[RTW-95]: https://warthogs.atlassian.net/browse/RTW-95?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,41 +76,32 @@ The terraform juju provider is documented over here: https://registry.terraform.
 
 Terraform tracks its state with a .tfstate file which is created as a result of running `terraform apply` -- for production purposes this will be stored in an S3-like bucket remotely, and for local development purposes it sits in the `terraform` directory aftery you have done a `terraform apply`).
 
-You can optionally get SSL certificates automatically managed for the ingress (in case you happen to have a DNS zone with Cloudflare DNS available):
-
-```bash
-TF_VAR_environment=development TF_VAR_external_ingress_hostname="mah-domain.com" TF_VAR_cloudflare_acme=true TF_VAR_cloudflare_dns_api_token=... TF_VAR_cloudflare_zone_read_api_token=... TF_VAR_cloudflare_email=... terraform apply -auto-approve
-```
-
 After all is up, you can run `juju switch test-observer-development` to use the development juju model. Then `juju status --relations` should give you output to the direction of the following (the acme-operator only there if `TF_VAR_cloudflare_acme` was passed in):
```
Comment (plars) [inline README.md:79]: Is this still valid? I don't see anything about TF_VAR_cloudflare_acme getting used in the tf plan.

**Review by plars — APPROVED** [review]
One minor question below, but I was able to deploy it using your instructions into my dev environment and it worked! The only odd thing I noticed was that after deploying and connecting to the frontend, I saw: `Error: DioException [connection error]: The connection errored: The XMLHttpRequest onError callback was called. This typically indicates an error on the network layer.` on the page toward the bottom. Likely that I'm just missing something with flutter maybe? Otherwise, I see no reason to delay landing this since it's been deployed successfully to staging. We can keep working on it from here and the changes should be a lot smaller :)

File: README.md (hunk 1 above)
Comment (omar-selo) [reply README.md:79]: Good catch. Also thanks for pointing out on mm that we don't need to setup GH credentials. I'll update the README

**Review by plars — APPROVED** [review]
+1

**Pull request merged**